### PR TITLE
CI: Test against Python 3.7

### LINKED ISSUE #20552: TST/COMPAT: test Python 3.7
Python 3.7.0b3 is released, it would be good to test against it.

Not sure what the best way is. Anaconda/conda-forge does not seem to have packages for 3.7, travis has a 3.7-dev python (then we need to switch our dev build from conda to pip).

### PR DESCRIPTION
**Pull request by TomAugspurger**
- [x] closes #20552 
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: ci/travis-37.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+name: pandas
+channels:
+  - defaults
+  - conda-forge
+  - c3i_test
```
Comment (TomAugspurger) [inline ci/travis-37.yaml]: @mingwandroid do you know if `c3i_test` will keep these 3.7 packages around until they're in the defaults?

**Comment by mingwandroid** [comment]
I will aim to keep as much there as possible.

**Comment by jreback** [comment]
can u update install.rst and setup.py where needed

**Comment by TomAugspurger** [comment]
Merging this, to ensure things don't slip in before the release.

**Comment by alimcmaster1** [comment]
Hey, @TomAugspurger 
I see two separate PR failing CI testing running Python3.7 due to the same error:
https://travis-ci.org/pandas-dev/pandas/jobs/396604150
https://travis-ci.org/pandas-dev/pandas/jobs/396594344

Is this expected? 

Thanks,

Alistair

**Pull request merged**